js/build.js

```diff
@@ -1,20 +1,36 @@
 Fliplet.Widget.instance('image', function (data) {
-  var $el = $(this);
-  var img;
+
+  var canvas = this;
   var imageUrl = data.image && data.image.url;
 
+  if (!imageUrl) {
+    return;
+  }
+
   // Load images after DOM Load
-  if (imageUrl && !$el.attr('src')) {
-    Fliplet.Navigator.onReady().then(function () {
-      $el[0].src = imageUrl;
+  Fliplet.Navigator.onReady().then(function () {
+    var $placeholder = $(canvas);
+    var img = document.createElement('IMG');
+    var propsToCopy = ['style', 'className', 'width', 'height'];
+    propsToCopy.forEach(function(x){
+      if (canvas.hasOwnProperty(x)) {
+        img[x] = canvas[x];
+      }
     });
-  }
+    img.dataset.imageId = canvas.dataset.imageId;
+    img.classList.remove('lazy-placeholder');
+    var $img = $(img);
+    $img.on('load', function(){
+      $placeholder.replaceWith(this);
+      $(this).hide().fadeIn(200);
+    }).attr('src', imageUrl);
 
-  $el.click(function (event) {
-    event.preventDefault();
+    $img.on('click', function (event) {
+      event.preventDefault();
 
-    if (data.action) {
-      Fliplet.Navigate.to(data.action);
-    }
+      if (data.action) {
```

Comment:
this can be swapped the other way around:

```
if (data.action) {
  $img.on('click', ...
}
```

don't you think? (unless we need the preventDefault anyway)